krabs/krabs/schema_locator.hpp

```diff
@@ -135,7 +131,6 @@ namespace krabs {
 
     private:
         static schema_locator singleton_;
-        critical_section sync_;
         std::unordered_map<schema_key, std::unique_ptr<char[]>> cache_;
```

Comment:
Make this mutable so the whole class can be used in const context.
Make the schema_locator::get_event_schema() const too.

```suggestion
        mutable std::unordered_map<schema_key, std::unique_ptr<char[]>> cache_;
```

Reply:
TIL: C++ has a mutable keyword.  :-)